cores/arduino/WInterrupts.c

```diff
@@ -132,25 +160,37 @@ void detachInterrupt(uint32_t pin)
   
   // Disable wakeup capability on pin during sleep
   EIC->WAKEUP.reg &= ~(1 << in);
+
+  //and take it out of the ISR list
+  int id=-1; //find its location first
+  for (uint32_t i=0; i<nints; i++){
+      if (ISRlist[i]==in) id=in;
+  }
+  if (id==-1) return; //apparently we didn't have it
+  for (uint32_t i=id; i<nints-1; i++){ //and shift the ones above one down
+      ISRlist[i]=ISRlist[i+1];
+      ISRcallback[i]=ISRcallback[i+1];
+  }
+  //and remove the top item
+  ISRlist[nints]=0;
+  ISRcallback[nints]=NULL;
```

Comment:
Is erasing the list element really needed? IMHO just decreasing `nints` should be enough, new elements added later will overwrite the old value.